server/api/person/__tests__/person.spec.js

```diff
@@ -193,7 +193,7 @@ test.serial('Should load a person into the db but block access and delete them v
   t.is(queriedPerson.phone, p.phone)
 })
 
-test.serial('Should find a person by email', async t => {
+test.serial('Should be block to find a person by email when not authorized', async t => {
```

Comment:
Rather than change this test you need to convert it to two tests.
1. find a person by email when authorized
2. block finding a person when not auth.

Otherwise the test will not cover the all the code.

Reply:
> I’m not saying you should hard code it - I’m saying that you generate the filter from the abilities and use it in the select request rather than post processing.
> […](#)
> On 10/10/2019, at 10:26, MinhPhu0304 ***@***.***> wrote: @MinhPhu0304 commented on this pull request. In server/api/person/person.controller.js <[#415 (comment)](https://github.com/voluntarily/vly2/pull/415#discussion_r333241195)>: > }) } catch (e) { console.error('Bad request', req.query) return res.status(400).send(e) } } +function removeField (request, queryResult) { That might not be a good idea. Because user should be able to see all their fields. By adding parameter, user's email and phone number + other user email or phone will be filtered out. That might be not good if user look in to their own profile page and their phone number is hidden — You are receiving this because your review was requested. Reply to this email directly, view it on GitHub <#415?email_source=notifications&email_token=AAMFYFOFD6AHLPJNXWYR4E3QNZEAVA5CNFSM4I65LRR2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCHOS5AI#discussion_r333241195>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAMFYFIU5NKD3VP4J3C53S3QNZEAVANCNFSM4I65LRRQ>.

But that will also filtered out user's record and other user's info when create the request